src/main/java/io/openliberty/tools/common/plugins/util/ServerFeatureUtil.java

```diff
@@ -380,18 +382,39 @@ private Properties getBootstrapProperties(File bootstrapProperties) {
     private String evaluateExpression(Properties properties, String expression) {
         String value = expression;
         if (expression != null) {
-            Pattern p = Pattern.compile("\\$\\{([^\\}]*)\\}");
+            Pattern p = Pattern.compile("\\$\\{(.*?)\\}");
             Matcher m = p.matcher(expression);
             StringBuffer sb = new StringBuffer();
             while (m.find()) {
                 String variable = m.group(1);
                 String propertyValue = properties.getProperty(variable, "\\$\\{" + variable + "\\}");
+                
+                // Remove encapsulating ${} characters and validate that a valid liberty directory property was configured
+                propertyValue = removeEncapsulatingEnvVarSyntax(propertyValue); 
+
                 m.appendReplacement(sb, propertyValue);
             }
             m.appendTail(sb);
             value = sb.toString();
         }
         return value;
     }
-    
+
+    private String removeEncapsulatingEnvVarSyntax(String propertyValue){
+        Pattern p = Pattern.compile("\\$\\{(.*?)\\}");
+        Matcher m = p.matcher(propertyValue);
+        StringBuffer sb = new StringBuffer();
+        while (m.find()) {
+            String envDirctoryProperty = m.group(1);
+            if(!VALID_LIBERTY_DIRECTORY_PROPERTY.contains(envDirctoryProperty)) {
+                warn("The directory property " + envDirctoryProperty + "specified is not a predifined Liberty directory property");
+                return null;
+            }
+            else {
+                m.appendReplacement(sb, envDirctoryProperty);
```

Comment:
What if the referenced variable is another bootstrap property? Probably should handle resolving that also.